include/mbedtls/ecjpake.h

```diff
@@ -44,6 +44,8 @@
 #include "ecp.h"
 #include "md.h"
 
+#define MBEDTLS_ERR_ECJPAKE_HW_ACCEL_FAILED               -0x0078  /**< ECJPAKE hardware accelerator failed */
```

Comment:
Treating ECJPAKE as a low-level module it inconsistent with the fact that it's built on top of ECP which is considered a high-level module (due to the fact that it build on top of MPI I guess).

Now I realise that in practice that distinction mostly matters because so that we can do this in high-level modules:
```
ret = some_low_level_function();
if( ret != 0 )
    ret += my_high_level_code;
```
and in practice I don't see than happening here, but it still bothers me to have a low-level module based on a "high-level" one.

Now I also realise we don't have enough spots left for high-level module any more, and so far we've cheated by deciding that modules based on ECP use error codes in the ECP namespace (which is why we have such things as `MBEDTLS_ERR_ECP_VERIFY_FAILED` rather than having it in `ECDSA`. So far most errors returned by ECJPAKE functions live in the ECP namespace too.

So I understand there may not be any good solution here in our current system, but I'm not really happy with adding low-level error codes fro ECJPAKE.

Reply:
Note: what I'm suggesting is that alternative implementation of ECJPAKE use the ECP error code, just like alt ECDH implementations do.

Reply:
Ugh. That's a good point. Indeed, it's annoying. @hanno-arm ECJPAKE_HW_ACCEL_FAILED is to complement #1304 . What do you think of Manuel's proposal to use ECP_HW_ACCEL_FAILED?

Reply:
@gilles-peskine-arm @mpg I support Manuel's proposal of using `ECP_HW_ACCEL_FAILED`. I think that while perhaps not optimal, having modules based on ECP use error codes from the ECP namespace -- thereby effectively collapsing the namespaces from there because our allocation of error codes doesn't allow to step 'one level up' -- is still cleaner than using low-level codes.

Reply:
@sbutcher-arm verbally approved using `ECP_HW_ACCEL_FAILED` for ECJPAKE, so I'll withdraw the `ECJPAKE_HW_ACCEL_FAILED` error code.